### DISCUSSION TITLE
Issues with base features

### DISCUSSION THREAD

**Discussion by ccrvlh** [microsoft/pylance-release#3352] [Q&A] [answered]
For the past month or so, there has been a significant decrease in the usefulness of tools available.
There are a few things there are really significant, but a lot more has happened.
I started noticing it more often from 1-2 months from now, the codebase is fairly small (~30k SLOC), noticed it happens less often with 2-3 files projects, but still not very reliable.

**1. Find references** 
Doesn't really find all references, only a couple are shown, even if open files of where the object is being referenced. 

**2. Rename**
While using F2 in won't rename every instance, this is not only true for edge cases (multiple files and references), but even for trivial cases (renaming params of a function for example). This seems to happen more often when this is done quickly, when you save right after changing the name, but I also experienced it without any action right after renaming things.

**3. Import suggestions**
This is really bad lately. It basically won't suggest anything, or just plain wrong suggests (like, far off).
I imagine this can be tricky for libraries with really nested objects, but it won't even suggest imports for sibling files, not even when both files (the one I'm coding on and the one that has the object that I need to import). This definitely got very bad to the point where it just doesn't work, start noticing it about 1-2 months ago, 2-4 weeks it just stopped working.

**4. Autocomplete**
Suggestions are really not useful: in one scenario it just doesn't show the correct package/module/param, in others is the second/third on the list, even though the word I'm writing is more similar to the 2/3rd item then the first.

**5. New Files**
New files are not recognized by the language server, only after the LS is restarted it recognizes the imports from new files.

**6. Change File Names**
When changing a file name, Pylance won't recognize that the file was changed and it won't update the imports.

I have the `analysis.index` to `True`, have fairly standard extensions (Docker / Toml / Remote / GraphQL / React / Remote), and also tried disabling most of them and keeping mostly Pylance and Intellicode. Tried with and without intellicode.

I'm running on a `macOS 11.6.8` using Pylance `2022.9.20` and Intellicode `1.2.27`.
Some of my settings are:
```json
  "python.languageServer": "Pylance",
  "python.formatting.provider": "black",
  "python.formatting.blackArgs": ["--line-length=100"],
  "python.linting.lintOnSave": true,
  "python.linting.enabled": true,
  "python.linting.flake8Enabled": true,
  "python.linting.mypyEnabled": true,
  "python.linting.pylintEnabled": false,
  "python.linting.ignorePatterns": [".venv/**", "migrations/**", "notebooks/**"],
  "python.diagnostics.sourceMapsEnabled": true,
  "python.analysis.typeCheckingMode": "off"
  "python.analysis.indexing": true,
  "python.pylanceLspNotebooksEnabled": false,
  "mypy.runUsingActiveInterpreter": true,
```
I have tried running a few keys on/off from the settings.json, but things doesn't seem to workout correcly..
The only logs I've are mostly related to application logic, so no specific bug I could see on the developer tools.
I can't replicate this in a micro project, so I have no idea what might be going wrong.

Ideas on what settings I might need to tweak to further debug this?
I'll spend some time trying to understand whats going on, but it does seem to be some sort of issue. I just performed a completely clean install of VSCode (deleting all extensions, cache folder, and downloading it all over again), started fresh only with the language server and the icons/theme, and same behavior). I'll try to replicate one by one to see if I can find a pattern that can be converted to an issue / feature request.

**Comment by ccrvlh** [comment]
# Test 1
1. New project created with poetry (`poetry init`)
2. New env created with poetry (`poetry shell` creates a venv in project, Python 3.10)
3. Create `test1` folder with `__init__.py`,  `schemas.py` and `services.py`
```python
# schemas.py
from dataclasses import dataclass

@dataclass
class MySchema:
    name: str
    age: int
```

```python
# services.py

# Manually add the import (after trying the autocomplete / quickactions didn't work)
from .schemas import MySchema


## When writing this function no autocomplete suggestion to import MySchema from schemas.py
## After writing no quick actions suggestion to import MySchema
def func(param: MySchema):
    pass
```

1. Find References (:white_check_mark: replicated)
This can be replicated adding a `nested` folder inside `test1`
Then adding a function that also uses `MySchema`. When in `schemas.py` and right click + Find References, it won't show the reference to `MySchema` being used by the func in `test1/nested/services.py`

2. Rename (:white_check_mark: replicated)
Pressing F2 on `MySchema` and changing it to `MySchema2` won't change the import to `from .schemas import MySchema2` and the `param: MySchema2`

3/4. Import Suggestions & Autocomplete (:white_check_mark: replicated)
While writing the function no suggestions to import `MySchema`
Quick actions (Cmd + `.`) also won't suggest the import after `MySchema` was manually added.

5. New Files (:white_check_mark: replicated)
After this is set up, create a new file called `new.py` with another dataclass.
Import the dataclass on the services file `from .new import NewDataclass`.
`from .new` will show an error: `Import ".new" could not be resolvedPylance`

6. Change file names (:white_check_mark: replicated)
Update the name of the file `schemas.py` to `new_schema.py`.
Imports on `services.py` won't change the name.

I wonder if any of these are by design, or if there's some sort of problem with my setup, or maybe Pylance is not able to work on any of these cases and it was all Copilot responsability (I was using it on the preview, and stopped using when the billing started), or if it's a bug.

**Comment by debonte** [comment, marked as answer]
@lowercase00, I converted this into an issue -- https://github.com/microsoft/pylance-release/issues/3363

Unfortunately, I'm unable to reproduce any of these issues, so we'll probably have some followup questions for you.
